### PR TITLE
Feat/roka credentials

### PR DESCRIPTION
**Pull request by RobusK**

### CODE REVIEW

File: src/main/java/com/webservices/controllers/OrderController.java [hunk 1]
```diff
@@ -23,11 +24,21 @@ public OrderController(OrderService orderService) {
...
+    }
+
     @GetMapping(value = "/{id}")
-    public ResponseEntity<?> getOrder(@PathVariable Integer id) {
+    public ResponseEntity<?> getOrder(@PathVariable Integer id, Authentication authentication) {
         try {
+            UserAuthenticationDecorator user = getUser(authentication);
             Order order = orderService.getOrder(id);
-            if (order == null) {
+            if (order == null || !order.getCustomer().getId().equals(user.getCustomer().getId())) {
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/controllers/OrderController.java]: use `doCustomerOwnOrder`

File: src/main/java/com/webservices/controllers/OrderController.java [hunk 2]
```diff
@@ -23,11 +24,21 @@ public OrderController(OrderService orderService) {
         this.orderService = orderService;
     }
 
+    private boolean doCustomerOwnOrder(int customerId, int orderId) {
+        Order order = this.orderService.getOrder(orderId);
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/controllers/OrderController.java]: can't we just compare two `Customer` instances

File: src/main/java/com/webservices/models/Customer.java [hunk 3]
```diff
@@ -12,33 +12,36 @@
 @Entity
 @Table(name = "CUSTOMERS")
 public class Customer {
-
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/models/Customer.java]: ugly

File: src/main/java/com/webservices/models/Supplier.java [hunk 4]
```diff
@@ -56,12 +68,12 @@ public void setProducts(List<Product> products) {
         this.products = products;
     }
 
-    public void fulfillParcel(Parcel parcel){
+    public void fulfillParcel(Parcel parcel) {
         List<Product> products = parcel.getProducts();
         List<Product> fulfilledProducts = new ArrayList<Product>();
-        for(Product product : products){
+        for (Product product : products) {
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/models/Supplier.java]: beautiful

File: src/main/java/com/webservices/models/User.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+import javax.persistence.*;
+import java.util.HashSet;
+import java.util.Set;
+
+@Entity
+@Table(name = "USERS")
+public class User {
+
+    private String username;
+    private String password;
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/models/User.java]: should be `char[]` but we can let it go

File: src/main/java/com/webservices/repositories/OrderDAOImpl.java [hunk 6]
```diff
@@ -90,6 +92,26 @@ public void cancelOrder(Integer orderId) {
...
+    public List<Order> getAllOrdersFromCustomer(Integer customerId) {
+        Session session = sessionFactory.openSession();
+        List<Order> allOrders = null;
+        try {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Order> criteriaQuery = criteriaBuilder.createQuery(Order.class);
+            Root<Order> root = criteriaQuery.from(Order.class);
+            criteriaQuery
+                    .select(root)
+                    .where(criteriaBuilder.equal(root.get("customer"), customerId));
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/repositories/OrderDAOImpl.java]: just to double check, should it be `"customer"` ?

File: src/main/java/com/webservices/repositories/UserDAOImpl.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                    .select(root)
+                    .where(criteriaBuilder.equal(root.get("username"), username));
+            users = session.createQuery(criteriaQuery).getResultList();
+
+            if (users.size() > 0) {
+                return users.get(0);
+            } else {
+                return null;
+            }
+        } finally {
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/repositories/UserDAOImpl.java]:
```
if(users.isEmpty()) {
    throw new Exception("no bueno);
}
catch(Exception e) {
   // logging
   return null
} finally {
  session.close();
  return users.get(0);
}

Comment (mDemianchuk) [reply src/main/java/com/webservices/repositories/UserDAOImpl.java]: actually throw usernotfound exception

Comment (RobusK) [reply src/main/java/com/webservices/repositories/UserDAOImpl.java]: Gets thrown on another level

File: src/main/java/com/webservices/repositories/OrderDAOImpl.java (hunk 6 above)
Comment (RobusK) [reply src/main/java/com/webservices/repositories/OrderDAOImpl.java]: Yes

File: src/main/java/com/webservices/controllers/OrderController.java (hunk 1 above)
Comment (RobusK) [reply src/main/java/com/webservices/controllers/OrderController.java]: Would result in double requests

File: src/main/java/com/webservices/repositories/UserDAOImpl.java (hunk 7 above)
Comment (mDemianchuk) [reply src/main/java/com/webservices/repositories/UserDAOImpl.java]: then 
```
User user = null;
List<User> users...
...
if(!users.isEmpty()) {
   user = users.get(0);
}
finally {
  session.close();
  return user;
}

**Pull request merged**